### ISSUE TITLE
PHP 8.1 Enumerations do not seem to work

### ISSUE THREAD

**Issue by TomOudeRengerink** [silverstripe/silverstripe-framework#10413]
## Affected Version
4.11.4

## Description

PHP 8.1 Enumerations do not seem to work (https://www.php.net/manual/en/language.enumerations.overview.php).

It throws `[Emergency] Uncaught Error: Class <EnumClass>` not found. Seems like it's looking for a class instead of a enum.

## Steps to Reproduce

1. Create a Enum

```
<?php

namespace App\Enums;

enum MyEnum
{
    case HELLO;
}
```

2. Import the enum and call the `cases` method.

```
use App\Enums\MyEnum;

MyEnum::cases();
```

**Comment by kinglozzer** [comment]
Hmm, my best guess without testing is that it’s not being included in our autoloader. Does it work if you add the App namespace to your composer.json autoloaders? e.g.

```json
{
    "autoload": {
        "psr-4": {
            "App\\": "app/src"
        }
    }
}
```

and run `composer dump-autoload`

Your enum would need to live in `app/src/Enums/MyEnum.php` to be PSR-4 compatible

**Comment by TomOudeRengerink** [comment]
@kinglozzer Yes this fixes the issue indeed! Is there a way to implement this in the class loader of SilverStripe somehow?

**Comment by kinglozzer** [comment]
I think it’ll be possible yes, it’ll mean some additions to `ClassManifest` as it currently looks at classes/traits/interfaces, but not enums

**Comment by kinglozzer** [comment]
First cut at adding support https://github.com/silverstripe/silverstripe-framework/pull/10415

**Comment by kinglozzer** [comment]
#10415 has been merged, so this will be released with 4.11. If anyone encounters this in the interim, the workaround is to use composer’s autoloader either by making your code PSR-4 compliant, or just by passing the path for the specific enum file(s):

```json
{
    "autoload": {
        "files": [
            "app/src/Path/To/Enum.php"
        ],
    }
}
```

(and then run `composer dump-autoload`)
